scripts/install-golang.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the
+# "License"). You may not use this file except in compliance
+#  with the License. A copy of the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and
+# limitations under the License.
+
+architecture=""
+case $(uname -m) in
+    x86_64)
+        architecture="amd64"
+        ;;
+    arm64)
+        architecture="arm64"
+        ;;
+    aarch64)
+        architecture="arm64"
+        ;;
+    *)
+        echo $"Unknown architecture $0"
+        exit 1
+esac
+
+if [ "$architecture" == "amd64" ]; then export GOARCH=amd64; fi
+if [ "$architecture" == "arm64" ]; then export GOARCH=arm64; fi
+
+# if golang isn't installed then we'll install it
+go_bin_path=$(which go || "")
+if [ $go_bin_path == "/usr/bin/go" ]; then
+    # rpm installs golang to the go_bin_path
+    echo "golang exists on /usr/bin/go, using system golang version"
+elif [ ${go_bin_path:0:10} == "/usr/local" ]; then
+    # using existing installed golang; re-export path to be sure it's there
+    export GOROOT=/usr/local/go
+    export PATH=$PATH:$GOROOT/bin
+    echo "$GOROOT exists. Using installed golang version"
+else
+    # install golang defined in GO_VERSION file
+    echo "$GOROOT doesn't exist, installing $(cat ./GO_VERSION)"
+    GO_VERSION=$(cat ./GO_VERSION)
+    tmpdir=$(mktemp -d)
+    GOLANG_TAR="go${GO_VERSION}.linux-${GOARCH}.tar.gz"
+    wget -O ${tmpdir}/${GOLANG_TAR} https://storage.googleapis.com/golang/${GOLANG_TAR}
```

Comment:
Could we use `curl` instead of `wget` for this? IME curl is almost always included by default on OSes and wget requires installation.

Reply:
```
++ curl https://storage.googleapis.com/golang/go1.19.1.linux-amd64.tar.gz -o /tmp/tmp.WfvR90vbnW/go1.19.1.linux-amd64.tar.gz
--
930 | ./scripts/install-golang.sh: line 54: curl: command not found
```

Updated PR and tested without `wget` -- looks like we'll need to install `curl` for debian builds.  I'll prefer wget in this case as it's already working and tested for all build types.  (We can revisit this in future if we have trouble.)